corehq/apps/es/transient_util.py

```diff
@@ -0,0 +1,103 @@
+"""Transient utilities needed during the interim of converting HQ Elastic logic
+to use the new "client adapter" models.
+"""
+
+from .client import ElasticDocumentAdapter
+from .registry import get_registry
+
+
+def get_adapter_mapping(adapter):
+    """Temporary function for fetching the Elastic mapping (still defined in
+    pillowtop module) for an adapter.
+    """
+    return _DOC_MAPPINGS_BY_INDEX[(adapter.index_name, adapter.type)]
+
+
+def from_dict_with_possible_id(doc):
+    """Temporary "common" function for adapters who don't yet own their document
+    ``from_python()`` logic.
+    """
+    if "_id" in doc:
+        return doc["_id"], {k: doc[k] for k in doc if k != "_id"}
```

Comment:
Using `del` would mutate the doc, which may be unexpected by calling code. It's a best practice to have functions not mutate arguments passed to them. Or if that is done it should be explicitly documented. I'm not sure if it makes sense to do that in this case? Like, is this always called in a context where the original `doc` dict is thrown away? If yes, then mutating it may be fine?

Reply:
That is a great point. I hadn't considered if this doc was used elsewhere. My understanding is that it should not be, but if that is not something we can enforce or verify, it may be preferable to keep the existing behavior.

Reply:
Yes, avoiding mutation was the goal here.  In the "transient" period of time this function will exist (until the adapters all own their respective `from_python()` logic), this function gets called on documents in many places, similarly to how/where the old `ElasticsearchInterface._without_id_field(doc)` method [did something similar](https://github.com/dimagi/commcare-hq/blob/252e729fda8/corehq/util/es/interface.py#L82-L86).